server/src/main/java/org/elasticsearch/index/mapper/MapperService.java

```diff
@@ -389,6 +389,10 @@ public MappedFieldType fieldType(String fullName) {
         return mappingLookup().fieldTypes().get(fullName);
     }
 
+    public MappedFieldType concreteFieldType(String field) {
+        return mappingLookup().fieldTypes().getConcrete(field);
+    }
```

Comment:
the fact that the distinction leaks into MapperService and we need this new method is concerning to me, maybe we should see if we can avoid this? otherwise it feels like we are introducing special logic for aliases, similar to what we already have for them in other places?